addons/stock/models/stock_move.py

```diff
@@ -247,10 +247,19 @@ def _get_move_lines(self):
 
     @api.depends('move_line_ids.qty_done', 'move_line_ids.product_uom_id')
     def _quantity_done_compute(self):
+        """ This field represents the sum of the move lines `qty_done`. It allows the user to know
+        if there is still work to do.
+
+        We take care of rounding this value at the general decimal precision and not the rounding
+        of the move's UOM to make sure this value is really close to the real sum, because this
+        field will be used in `_action_done` in order to know if the move will need a backorder or
+        an extra move.
+        """
         for move in self:
+            quantity_done = 0
             for move_line in move._get_move_lines():
-                # Transform the move_line quantity_done into the move uom.
-                move.quantity_done += move_line.product_uom_id._compute_quantity(move_line.qty_done, move.product_uom)
+                quantity_done += move_line.product_uom_id._compute_quantity(move_line.qty_done, move.product_uom, round=False)
+            move.quantity_done = float_round(quantity_done, precision_digits=self.env['decimal.precision'].precision_get('Product Unit of Measure'))
 
     def _quantity_done_set(self):
```

Comment:
issue here: with this patch, it is possible to set 0.5 even if the move's uom has a rounding of 1.0